shared/yaml/validation.py

```diff
@@ -61,6 +61,11 @@ def pre_process_yaml(inputted_yaml_dict):
             val = inputted_yaml_dict["codecov"]["notify"].pop("require_ci_to_pass")
             inputted_yaml_dict["codecov"]["require_ci_to_pass"] = val
 
+    # Remove the reserved `to_string` key word in the base level of the json.
+    # This key is used to store the full YAML string preserving the original comments.
+    if "to_string" in inputted_yaml_dict:
+        inputted_yaml_dict.pop("to_string")
```

Comment:
Thoughts on adding a helper `remove_reserved_keys` that lets us pop any reserved data in the future?

Something like below?

```

yaml_top_level_reserved_keys: List[str] = ["to_string"]

def remove_reserved_keys(input_dict: Dict[str, any], keys_to_remove: List[str]) -> Dict[str, any]:
    copied_dict = copy.deepcopy(input_dict)
    
    for key in keys_to_remove:
        if key in copied_dict:
            copied_dict.pop(key)
    
    return copied_dict
```